client/gutenberg/extensions/map/component.js

```diff
@@ -0,0 +1,338 @@
+		const { api_key, children, points, map_style, map_details } = this.props;
+		const { map } = this.state;
+		if ( api_key && api_key.length > 0 && api_key !== prevProps.api_key ) {
+			window.mapboxgl = null;
+			this.loadMapLibraries();
+		}
+		// If the user has just clicked to show the Add Point component, hide info window.
+		// AddPoint is the only possible child.
+		if ( children !== prevProps.children && children !== false ) {
+			this.clearCurrentMarker();
+		}
+		if ( points !== prevProps.points ) {
+			this.setBoundsByMarkers();
+		}
+		if ( points.length !== prevProps.points.length ) {
+			this.clearCurrentMarker();
+		}
+		if ( map_style !== prevProps.map_style || map_details !== prevProps.map_details ) {
+			map.setStyle( this.getMapStyle() );
+		}
+	}
+	/* Event handling */
+	onMarkerClick = marker => {
+		const { onMarkerClick } = this.props;
+		this.setState( { activeMarker: marker } );
+		onMarkerClick();
+	};
+	onMapClick = () => {
+		this.setState( { activeMarker: null } );
+	};
+	clearCurrentMarker = () => {
+		this.setState( { activeMarker: null } );
+	};
+	updateActiveMarker = updates => {
+		const { points } = this.props;
+		const { activeMarker } = this.state;
+		const { index } = activeMarker.props;
+		const newPoints = points.slice( 0 );
+
+		assign( newPoints[ index ], updates );
+		this.props.onSetPoints( newPoints );
+	};
+	deleteActiveMarker = () => {
+		const { points } = this.props;
+		const { activeMarker } = this.state;
+		const { index } = activeMarker.props;
+		const newPoints = points.slice( 0 );
+
+		newPoints.splice( index, 1 );
+		this.props.onSetPoints( newPoints );
+		this.setState( { activeMarker: null } );
+	};
+	// Various map functions
+	sizeMap = () => {
+		const { map } = this.state;
+		const mapEl = this.mapRef.current;
+		const blockWidth = mapEl.offsetWidth;
+		const maxHeight = window.innerHeight * 0.8;
+		const blockHeight = Math.min( blockWidth * ( 3 / 4 ), maxHeight );
+		mapEl.style.height = blockHeight + 'px';
+		map.resize();
+		this.setBoundsByMarkers();
+	};
+	setBoundsByMarkers = () => {
+		const { zoom, points, onSetZoom } = this.props;
+		const { map, activeMarker, mapboxgl, zoomControl, fit_to_bounds } = this.state;
+
+		const bounds = new mapboxgl.LngLatBounds();
+		if ( ! map || ! points.length || activeMarker ) {
+			return;
+		}
+		points.forEach( point => {
+			bounds.extend( [ point.coordinates.longitude, point.coordinates.latitude ] );
+		} );
+		map.setCenter( bounds.getCenter() );
+
+		// If there are multiple points, zoom is determined by the area they cover,
+		// and zoom control is removed.
+		if ( points.length > 1 ) {
+			map.fitBounds( bounds, {
+				padding: {
+					top: 40,
+					bottom: 40,
+					left: 20,
+					right: 20,
+				},
+			} );
+			this.setState( { fit_to_bounds: true } );
+			map.removeControl( zoomControl );
+			return;
+		}
+		/* Case where points go from multiple to just one. Set zoom to an arbitrarily high level. */
+		if ( fit_to_bounds ) {
```

Comment:
Since we call `map.fitBounds()` regardless of `fit_to_bounds`, should we call `fit_to_bounds` something else?

Reply:
Would like to look at this as part of a broader refactor of `setBoundsByMarkers()`, post-merge.